### PR TITLE
Add verification data export page in stats module

### PR DESCRIPTION
**Pull request by mllocs**
#### :tophat: What? Why?

On some occasions, it is necessary to know the sociodemographic data of the participants in a process. Currently, this data can be obtained from the verification form. That is, if you ask for the date of birth, gender, and postal code during the verification, you can then aggregate this data to know the sociodemographic profile of the participants.

#### :pushpin: Related Issues

- Fixes https://github.com/AjuntamentdeBarcelona/decidim-barcelona/issues/548

### :camera: Screenshots (optional)

Screenshot 2024-10-04 at 11 59 58

### CODE REVIEW

**Comment by mllocs** [comment]
I managed to add the `decidim-stats` module to our decidim-lot2-staging server. You can find the new export page here https://decidim-lot2.populate.tools/stats/authorization_exports

@furilo There might not be enough verifications data to export, do you want me to create some?

**Review by furilo — DISMISSED** [review]

File: decidim-stats/config/locales/ca.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+ca:
+  decidim:
+    stats:
+      authorization_exports:
```
Comment (ferblape) [inline decidim-stats/config/locales/ca.yml:4]: @mllocs translations are pending? All copied from English

**Review by ferblape — APPROVED** [review]
Apart from the translations, the rest looks ok to me

File: decidim-stats/config/locales/ca.yml (hunk 1 above)
Comment (mllocs) [reply decidim-stats/config/locales/ca.yml:4]: oh, right. I can do the translations.

Comment (mllocs) [reply decidim-stats/config/locales/ca.yml:4]: Done

**Comment by NilHomedes** [comment]
Hi @mllocs 

>To fully test this we could try to configure your authorization method to our staging or, in case you have a staging server, we could deploy the export data page there.

Yes, that could be the best way to test this module. Take into account that Barcelona is using different authorizations handlers:
- https://github.com/AjuntamentdeBarcelona/decidim-barcelona/tree/master/decidim-census_sms
- https://github.com/AjuntamentdeBarcelona/decidim-barcelona/blob/master/app/services/census_authorization_handler.rb


---
By the way, I leave a couple of things I have seen:

- [ ] 1) To comply with Decidim Design system, we should add an icon to the side menu element “Export data”. 

2) It seems that is not taking the birthday data. As you can see, the “Example Authorization” has a birthday filed, but it’s empty when exporting the data. 

imatge

So the question is: This module would work for any type of authorizations handler? Or it will only works for Barcelona's authorizations?

**Comment by mllocs** [comment]
> * [ ]  1) To comply with Decidim Design system, we should add an icon to the side menu element “Export data”.
> 

@furilo @NilHomedes which icon do you suggest to use for "Export data"? https://nightly.decidim.org/design/foundations/iconography

**Comment by furilo** [comment]
> > * [ ]   1) To comply with Decidim Design system, we should add an icon to the side menu element “Export data”.
> 
> @furilo @NilHomedes which icon do you suggest to use for "Export data"? https://nightly.decidim.org/design/foundations/iconography

I guess `download-line`

**Comment by mllocs** [comment]
> It seems that is not taking the birthday data. As you can see, the “Example Authorization” has a birthday filed, but it’s empty when exporting the data.

This is because the "Example Authorization" doesn't even store the birthday as a metadata 🤷  https://github.com/decidim/decidim/blob/develop/decidim-generators/lib/decidim/generators/app_templates/dummy_authorization_handler.rb#L72
It will work for the barcelona verification methods, I'll try to set them up in our staging... 🤞

**Comment by mllocs** [comment]
> So the question is: This module would work for any type of authorizations handler? Or it will only works for Barcelona's authorizations?

It will work for authorizations that store the metadata the same way Barcelona's authorizations do.

**Comment by NilHomedes** [comment]
@mllocs Thanks for the explanation, now it's clearer to me. 
In this case, I would wait until we can test this in Barcelona's environment

**Comment by mllocs** [comment]
@NilHomedes I tried it in decidim-barcelona's staging, this is what I found:

- Exporting "El padró" method worked well except for the "scope_name" column, I'm fixing that. 
- Exporting "Codi per SMS" didn't export any data, I wonder if It should work or if that method doesn't collect any kind of census data.

**Comment by mllocs** [comment]
@fblupi I added a commit fixing one little thing, could you redeploy please?

**Comment by fblupi** [comment]
@mllocs I've just deployed it

**Review by NilHomedes — APPROVED** [review]
Working as expected, thanks @mllocs 💯 

I'm approving this to move it forward. 
Just one suggestion, could you document this new feature in the Readme module?

**Review by fblupi — APPROVED** [review]

**Pull request merged**